modules/tensor_mechanics/src/scalarkernels/PeriodicStrain.C

```diff
@@ -0,0 +1,135 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "PeriodicStrain.h"
+
+// MOOSE includes
+#include "Assembly.h"
+#include "PeriodicStrainUserObject.h"
+#include "MooseVariableScalar.h"
+#include "SystemBase.h"
+#include "RankTwoTensor.h"
+#include "RankFourTensor.h"
+
+registerMooseObject("TensorMechanicsApp", PeriodicStrain);
+
+template <>
+InputParameters
+validParams<PeriodicStrain>()
+{
+  InputParameters params = validParams<ScalarKernel>();
+  params.addClassDescription("Periodic Strain Scalar Kernel");
+  params.addRequiredParam<UserObjectName>("periodic_strain",
+                                          "The name of the PeriodicStrainUserObject");
+
+  return params;
+}
+
+PeriodicStrain::PeriodicStrain(const InputParameters & parameters)
+  : ScalarKernel(parameters),
+    _pst(getUserObject<PeriodicStrainUserObject>("periodic_strain")),
+    _pst_residual(_pst.getResidual()),
+    _pst_jacobian(_pst.getJacobian()),
+    _components(_var.order())
+{
+  if (_var.order() != FIRST && _var.order() != THIRD && _var.order() != SIXTH &&
+      _var.order() != NINTH)
+    mooseError("PerdiodicStrain ScalarKernel is only compatible with FIRST, THIRD, SIXTH,"
+               "and NINTH order scalar variables. Please change order of the scalar"
+               "variable acoording to the problem dimension.");
```

Comment:
If you allow NINTH you also need to allow FOURTH. I'd rather not support NINTH for now.

Reply:
I was actually thinking of creating a case for FOURTH as well, then skipped it.